src/modals/import-account-modal/import-account-seed/import-account-seed.tsx

```diff
@@ -0,0 +1,102 @@
+import { FormPasswordInput } from '../../../form/form-password-input';
+import { FormRadioButtonsGroup } from '../../../form/form-radio-buttons-group';
+import { FormTextInput } from '../../../form/form-text-input';
+import { ImportAccountDerivationEnum, ImportAccountSeedValues } from '../../../interfaces/import-account-type';
+import { useShelter } from '../../../shelter/use-shelter.hook';
+import { useAccountsListSelector } from '../../../store/wallet/wallet-selectors';
+import { formatSize } from '../../../styles/format-size';
+import { getDerivationPath, seedToHDPrivateKey } from '../../../utils/keys.util';
+import { useImportAccountStyles } from '../import-account.styles';
+import { importAccountSeedInitialValues, importAccountSeedValidationSchema } from './import-account-seed.form';
+
+interface Props {
+  onBackHandler: EmptyFn;
+}
+
+export const ImportAccountSeed: FC<Props> = ({ onBackHandler }) => {
+  const styles = useImportAccountStyles();
+  const { createImportedAccount } = useShelter();
+  const accountsIndex = useAccountsListSelector().length + 1;
+
+  const onSubmit = (values: ImportAccountSeedValues) => {
+    const seed = mnemonicToSeedSync(values.seedPhrase);
+    const privateKey = seedToHDPrivateKey(
+      seed,
+      values.derivationType === ImportAccountDerivationEnum.DEFAULT ? getDerivationPath(0) : values.derivationPath
+    );
+    createImportedAccount({
+      name: `Account ${accountsIndex}`,
+      privateKey
+    });
+  };
+
+  return (
+    <Formik
+      initialValues={importAccountSeedInitialValues}
+      validationSchema={importAccountSeedValidationSchema}
+      enableReinitialize={true}
+      onSubmit={onSubmit}>
+      {({ values, submitForm, isValid }) => (
+        <ScreenContainer isFullScreenMode={true}>
+          <View>
+            <Divider size={formatSize(12)} />
+            <View style={styles.seedPhraseInputContainer}>
+              <Label label="Seed phrase" description="Mnemonic. Your secret 12 or more words." />
+              <FormMnemonicInput name="seedPhrase" />
+            </View>
+            <Divider size={formatSize(12)} />
+            <Label
+              label="Password"
+              isOptional
+              description={'Used for additional mnemonic derivation.\nThat is NOT a wallet password.'}
+            />
+            <FormPasswordInput name="password" />
+            <Divider size={formatSize(12)} />
+            <Label
+              label="Derivation"
+              isOptional
+              description="By default derivation isn't used. Click on 'Custom derivation path' to add it."
+            />
+            <FormRadioButtonsGroup
+              name="derivationType"
+              buttons={[
+                { value: ImportAccountDerivationEnum.DEFAULT, label: 'Default account (the first one)' },
+                { value: ImportAccountDerivationEnum.CUSTOM_PATH, label: 'Custom derivation path' }
+              ]}
```

Comment:
the same
https://github.com/madfish-solutions/templewallet-mobile/pull/172/files#r684012255